src/interface/mod.rs

```diff
@@ -0,0 +1,119 @@
+use alloy::{contract::Error, providers::RootProvider, sol, transports::http::Http};
+use reqwest::Client;
+use serde::{Deserialize, Serialize};
+
+use self::IAggregatorV3Interface::IAggregatorV3InterfaceInstance;
+
+sol!(
+    #[allow(missing_docs)]
+    #[sol(rpc)]
+    IAggregatorV3Interface,
+    "src/abi/IAggregatorV3Interface.json"
+);
+
+pub trait Serializable {
+    fn to_bin(&self) -> Result<Vec<u8>, Box<bincode::ErrorKind>>;
+    fn from_bin(data: Vec<u8>) -> Result<Self, Box<bincode::ErrorKind>>
+    where
+        Self: Sized;
+}
+
+#[derive(Clone)]
+pub struct ChainlinkContract {
+    pub contract: IAggregatorV3InterfaceInstance<Http<Client>, RootProvider<Http<Client>>>,
+    pub decimals: u8,
+}
+
+/// The latest price received for this symbol.
+/// This data is directly retrieved from the underlying contract.
+#[derive(Deserialize, Serialize)]
```

Comment:
Consider adding `Debug` to derive so that you can use `println!("{:#?}", price_data);` in tests to get formatted output.